balancer-labs/frontend-v2
# Feat: join/exit Aura swap trade

# Description

With the release of new Balanced SDK version, it is now possible to perform a new type of swap, "Join/Exit" swap. This uses batch relayer to perform multiple transactions which join and exit given pool.

Example of how to use this:
https://github.com/balancer-labs/balancer-sdk/blob/develop/balancer-js/examples/swapSor.ts

The swap type priority is this:
1. If user selected "Gasless" AKA "CoW swap" AKA "Gnosis" > use Gnosis trade.
2. If user unselected "Gasless" or it's not available > check if JoinExit is available and use it.
3. If JoinExit trade is not available > use SOR

Please see below on how to test.

## Type of change


## How should this be tested?

On Mainnet, test different trade types.
On Polygon, BAL -> AVAX swap should use JoinExit.
On Polygon, BAL -> MATIC swap should not use JoinExit.
Check other networks.

## Visual context

N/A

## Checklist:
